common/djangoapps/track/contexts.py

```diff
@@ -12,23 +12,32 @@
 log = logging.getLogger(__name__)
 
 
-def course_context_from_url(url):
+def course_context_from_url(url, course_id_string=None):
     """
-    Extracts the course_context from the given `url` and passes it on to
-    `course_context_from_course_id()`.
+    If course_id_string string is not present, extracts it from the given `url`. Either way, then passes
+    it on to `course_context_from_course_id()`.
     """
     url = url or ''
-
-    match = COURSE_REGEX.match(url)
     course_id = None
-    if match:
-        course_id_string = match.group('course_id')
+
+    if not course_id_string:
+        match = COURSE_REGEX.match(url)
+        if match:
+            course_id_string = match.group('course_id')
+    if not course_id_string:
+        log.warning(
+            'no course_id found in "{url}"'.format(
+                url=repr(url)[0:256]
+            ),
+            exc_info=True
+        )
```

Comment:
Not sure how often we expect to find a course_id embedded in our URLs, but there may be a fair number of warnings resulting from this.  (Maybe the other changes in this PR will reduce the number, but it may still be high.)

Reply:
would it make sense to change the logging level? maybe to debug instead of warning?

Reply:
Yes, I think debug instead of warning would be good.

Reply:
ok, doing that